opm/simulators/utils/SetupZoltanParams.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+  Copyright 2022 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/simulators/utils/SetupZoltanParams.hpp>
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+
+#include <boost/property_tree/json_parser.hpp>
+
+namespace Opm
+{
+
+
+std::map<std::string,std::string> setupZoltanParams(const std::string& conf)
+{
+    std::map<std::string,std::string> result;
+
+    if (conf == "scotch") {
+        result.insert({"LB_METHOD", "GRAPH"});
+        result.insert({"GRAPH_PACKAGE", "Scotch"});
+    } else if (conf == "hypergraph") {
+        result.insert({"LB_METHOD", "HYPERGRAPH"});
+    } else if (conf == "graph") {
+        result.insert({"LB_METHOD", "GRAPH"});
+        result.insert({"GRAPH_PACKAGE", "PHG"});
```

Comment:
Any reason not to use
```C++
result.emplace(key, value);
```
instead of
```C++
result.insert({key, value});
```
here?  The former constructs the `value_type` in place while the latter is a construct+move.

Reply:
no reason, changed.